### PR TITLE
Automation test for Death/Necropsy

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Automation test for Death/Necropsy. 

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: nirc_ehr/test/src/org.labkey.test/tests.nirc_ehr/NIRC_EHRTest.java [hunk 1]
```diff
@@ -364,54 +400,86 @@ private void createUser(String userEmail, String groupName, @Nullable String per
...
+        switchToWindow(1);
+        submitForm("Submit Final", "Finalize");
+        switchToMainWindow();
+        stopImpersonating();
+
+        log("Verify rows were inserted in appropriate datasets");
+        goToEHRFolder();
+        verifyRowCreated("study", "weight", aliveAnimalId, 1);
 
-        //verify that animal's record is closed, meaning 'end date' is set in study.assignments, study.protocolAssignments, and study.housing as the date of death
```
Comment (labkey-bpatel) [inline nirc_ehr/test/src/org.labkey.test/tests.nirc_ehr/NIRC_EHRTest.java:414]: I don't see this being verified. If not this time around, then add this comment as a TODO

Comment (labkey-sweta) [reply nirc_ehr/test/src/org.labkey.test/tests.nirc_ehr/NIRC_EHRTest.java:414]: There are no records in study.assignments, study.housing and study.protocolAssignments

Should i insert them before marking animal dead?

**Review by labkey-bpatel — APPROVED** [review]
Approving the test related changes.
Awaiting approval on the changes I made.

File: nirc_ehr/resources/scripts/nirc_triggers.js [hunk 2]
```diff
@@ -70,4 +70,10 @@ exports.init = function (EHR) {
             removeTimeFromDate: false,
         });
     });
+
+    EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.INIT, 'study', 'deaths', function(event, helper) {
```
Comment (ankurjuneja) [inline nirc_ehr/resources/scripts/nirc_triggers.js:74]: does this take precedence over the scriptOptions defined in deaths.js?

File: nirc_ehr/src/org/labkey/nirc_ehr/table/NIRC_EHRCustomizer.java [hunk 3]
```diff
@@ -42,13 +43,121 @@ public void customize(TableInfo table)
     {
         if (table instanceof AbstractTableInfo ti)
         {
+            sharedCustomization(ti);
             doSharedCustomization(ti);
```
Comment (ankurjuneja) [inline nirc_ehr/src/org/labkey/nirc_ehr/table/NIRC_EHRCustomizer.java]: looks like some columns like "performedBy" and "gender" are getting customized both in sharedCustomization & doSharedCustomization and customizations in sharedCustomization may get lost. Can these two methods be consolidated?

**Review by ankurjuneja — CHANGES_REQUESTED** [review]

File: nirc_ehr/resources/scripts/nirc_triggers.js (hunk 2 above)
Comment (labkey-bpatel) [reply nirc_ehr/resources/scripts/nirc_triggers.js:74]: Yes it does, the scriptOptions in deaths.js were getting overridden by the one in EHR, so ended up adding it here.

Comment (labkey-bpatel) [reply nirc_ehr/resources/scripts/nirc_triggers.js:74]: Removed the one in death.js

File: nirc_ehr/src/org/labkey/nirc_ehr/table/NIRC_EHRCustomizer.java (hunk 3 above)
Comment (labkey-bpatel) [reply nirc_ehr/src/org/labkey/nirc_ehr/table/NIRC_EHRCustomizer.java]: Good catch! now consolidated

**Review by ankurjuneja — APPROVED** [review]

**Pull request merged**